src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/FindLargestCrossSections.cpp

```diff
@@ -0,0 +1,112 @@
+
+using namespace complex;
+
+// -----------------------------------------------------------------------------
+FindLargestCrossSections::FindLargestCrossSections(DataStructure& dataStructure, const IFilter::MessageHandler& mesgHandler, const std::atomic_bool& shouldCancel,
+                                                   FindLargestCrossSectionsInputValues* inputValues)
+: m_DataStructure(dataStructure)
+, m_InputValues(inputValues)
+, m_ShouldCancel(shouldCancel)
+, m_MessageHandler(mesgHandler)
+{
+}
+
+// -----------------------------------------------------------------------------
+FindLargestCrossSections::~FindLargestCrossSections() noexcept = default;
+
+// -----------------------------------------------------------------------------
+const std::atomic_bool& FindLargestCrossSections::getCancel()
+{
+  return m_ShouldCancel;
+}
+
+// -----------------------------------------------------------------------------
+Result<> FindLargestCrossSections::operator()()
+{
+  const auto& imageGeom = m_DataStructure.getDataRefAs<ImageGeom>(m_InputValues->ImageGeometryPath);
+  auto& featureIds = m_DataStructure.getDataRefAs<Int32Array>(m_InputValues->FeatureIdsArrayPath);
+  auto& largestCrossSections = m_DataStructure.getDataRefAs<Float32Array>(m_InputValues->LargestCrossSectionsArrayPath);
+  const usize numFeatures = largestCrossSections.getNumberOfTuples();
+
+  // Validate the largestCrossSections array is the proper size
+  auto validateResults = ValidateNumFeaturesInArray(m_DataStructure, m_InputValues->LargestCrossSectionsArrayPath, featureIds);
+  if(validateResults.invalid())
+  {
+    return validateResults;
+  }
+
+  std::vector<float32> featureCounts(numFeatures, 0.0f);
+
+  usize outPlane = 0, inPlane1 = 0, inPlane2 = 0;
+  float32 resScalar = 0.0f, area = 0.0f;
+  usize stride1 = 0, stride2 = 0, stride3 = 0;
+  usize iStride = 0, jStride = 0, kStride = 0;
+  usize point = 0, gNum = 0;
+
+  FloatVec3 spacing = imageGeom.getSpacing();
+
+  if(m_InputValues->Plane == 0)
+  {
+    outPlane = imageGeom.getNumZCells();
+    inPlane1 = imageGeom.getNumXCells();
+    inPlane2 = imageGeom.getNumYCells();
+
+    resScalar = spacing[0] * spacing[1];
+    stride1 = inPlane1 * inPlane2;
+    stride2 = 1;
+    stride3 = inPlane1;
+  }
+  if(m_InputValues->Plane == 1)
+  {
+    outPlane = imageGeom.getNumYCells();
+    inPlane1 = imageGeom.getNumXCells();
+    inPlane2 = imageGeom.getNumZCells();
+    resScalar = spacing[0] * spacing[2];
+    stride1 = inPlane1;
+    stride2 = 1;
+    stride3 = inPlane1 * inPlane2;
+  }
+  if(m_InputValues->Plane == 2)
+  {
+    outPlane = imageGeom.getNumXCells();
+    inPlane1 = imageGeom.getNumYCells();
+    inPlane2 = imageGeom.getNumZCells();
+    resScalar = spacing[1] * spacing[2];
+    stride1 = 1;
+    stride2 = inPlane1;
+    stride3 = inPlane1 * inPlane2;
+  }
+  for(size_t i = 0; i < outPlane; i++)
+  {
+    for(size_t g = 0; g < numFeatures; g++)
+    {
+      featureCounts[g] = 0.0f;
+    }
```

Comment:
```suggestion
    std::fill(featureCounts.begin(), featureCounts.end(), 0.0f);

```
Just use fill to avoid extra `for` loop for readability